gtsam/hybrid/HybridValues.h

```diff
@@ -0,0 +1,127 @@
+/* ----------------------------------------------------------------------------
+
+ * GTSAM Copyright 2010, Georgia Tech Research Corporation,
+ * Atlanta, Georgia 30332-0415
+ * All Rights Reserved
+ * Authors: Frank Dellaert, et al. (see THANKS for the full author list)
+
+ * See LICENSE for the license information
+
+ * -------------------------------------------------------------------------- */
+
+/**
+ *  @file HybridValues.h
+ *  @date Jul 28, 2022
+ *  @author Shangjie Xue
+ */
+
+#pragma once
+
+#include <gtsam/discrete/Assignment.h>
+#include <gtsam/discrete/DiscreteKey.h>
+#include <gtsam/discrete/DiscreteValues.h>
+#include <gtsam/inference/Key.h>
+#include <gtsam/linear/VectorValues.h>
+#include <gtsam/nonlinear/Values.h>
+
+#include <map>
+#include <string>
+#include <vector>
+
+namespace gtsam {
+
+/**
+ * HybridValues represents a collection of DiscreteValues and VectorValues.  It
+ * is typically used to store the variables of a HybridGaussianFactorGraph.
+ * Optimizing a HybridGaussianBayesNet returns this class.
+ */
+class GTSAM_EXPORT HybridValues {
```

Comment:
I don't understand the use of this class. Can't we simply have `optimize` run MPE on the discrete tree and then run a regular optimization on the continuous values corresponding to the assignment? That way we can just return a `pair<Values, DiscreteValues>`.

Reply:
Yeah, I think that's also an option. But I feel like the api is not very user friendly while accessing/assigning values. We can talk about it in our meeting.